src/main.rs

```diff
@@ -61,3 +71,35 @@ enum VigSubCommands {
         query: Vec<String>,
     },
 }
+
+fn show_version() {
+    let config = Configuration::new();
+    let config_clone = config.clone();
+    let result = default_api::get_version(&config_clone);
+
+    let cli_name = crate_name!();
+    let cli_version = crate_version!();
+    // get the api version from the user agent
+    let user_agent = config.user_agent.unwrap();
+    let cli_api_version = user_agent.split("/").nth(1).unwrap();
+
+    println!("{cli_name} {cli_version}");
+    println!("API version: {cli_api_version}");
+
+    let rt = Runtime::new().unwrap();
+    rt.block_on(async {
```

Comment:
Fetching the service version will take the longest. Is there any way to make use of the async nature and start the fetch process as the very first step and then do the rest while "waiting" for a response. I understand that we want to have a deterministic way of printing the version, aka always the same order but thus should be possible by just not print the result as soon as we get it.
I know for all parts of the system are on the same host, this is not really a big improvement in any way, but I thought, I'll mention it anyway.